docs/workers.md

```diff
@@ -117,23 +117,23 @@ redis:
     enabled: true
 ```
 
-See the sample config for the full documentation of each option.
+See the [configuration manual](usage/configuration/config_documentation.html) for the full documentation of each option.
 
 Under **no circumstances** should the replication listener be exposed to the
 public internet; it has no authentication and is unencrypted.
 
 
 ### Worker configuration
 
-In the config file for each worker, you must specify the type of worker
-application (`worker_app`), and you should specify a unique name for the worker
-(`worker_name`). The currently available worker applications are listed below.
-You must also specify the HTTP replication endpoint that it should talk to on
-the main synapse process.  `worker_replication_host` should specify the host of
-the main synapse and `worker_replication_http_port` should point to the HTTP
-replication port. If the worker will handle HTTP requests then the
-`worker_listeners` option should be set with a `http` listener, in the same way
-as the `listeners` option in the shared config.
+In the config file for each worker, you must specify:
+ * The type of worker (`worker_app`). The currently available worker applications are listed below.
+ * A unique name for the worker (`worker_name`).
+ * The HTTP replication endpoint that it should talk to on the main synapse process
+   (`worker_replication_host` and `worker_replication_http_port`)
+ * If handling HTTP requests, a `worker_listeners` option with an `http`
+   listener, in the same way as the `listeners` option in the shared config.
+ * If handling the `^/_matrix/client/v3/keys/upload` endpoint, the HTTP URI for
```

Comment:
For my own edification-why is this required? (Not questioning, just curious)

Reply:
which bit, in particular?

Reply:
@H-Shay I guess you're asking about `worker_main_http_uri`? In which case, the answer is that `/_matrix/client/v3/keys/upload` is sometimes processed locally on the worker, but sometimes needs passing over to the main process; and the way it does the latter is very hacky and just an HTTP hit on the main process's regular HTTP interface.

https://github.com/matrix-org/synapse/issues/13642 is about improving this.

Reply:
That was it, thank you for explaining!